dnd5esheets/front/src/store/index.ts

```diff
@@ -114,6 +169,140 @@ class CharacterStore {
         })
       )
     }
+
+  private wireEffects = (character: ResolvedCharacter) => {
+    const abilityKeys = Object.keys(character.data.abilities) as Array<
+      keyof Abilities
+    >
+
+    const baseEffects = [
+      // proficiency bonus effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.proficiency_bonus := Math.ceil(1 + level / 4)`,
+          character
+        ),
+      },
+
+      // ability modifiers effects
+      ...abilityKeys.map((ability) => ({
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.abilities.${ability}.modifier := Math.floor((data.abilities.${ability}.score - 10) / 2)`,
+          character
+        ),
+      })),
+
+      // ability saves effects
+      ...abilityKeys.map((ability) => ({
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.abilities.${ability}.save := data.abilities.${ability}.modifier + data.abilities.${ability}.proficiency * data.proficiency_bonus`,
+          character
+        ),
+      })),
+
+      // skills effects
+      ...[
+        ['acrobatics', 'dexterity'],
+        ['animal_handling', 'wisdom'],
+        ['arcana', 'intelligence'],
+        ['athletics', 'strength'],
+        ['deception', 'dexterity'],
+        ['history', 'intelligence'],
+        ['insight', 'wisdom'],
+        ['intimidation', 'charisma'],
+        ['investigation', 'intelligence'],
+        ['medicine', 'wisdom'],
+        ['nature', 'intelligence'],
+        ['perception', 'wisdom'],
+        ['performance', 'charisma'],
+        ['persuasion', 'charisma'],
+        ['religion', 'intelligence'],
+        ['sleight_of_hand', 'dexterity'],
+        ['stealth', 'dexterity'],
+        ['survival', 'wisdom'],
+      ].map(([skill, ability]) => ({
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.skills.${skill}.modifier := data.abilities.${ability}.modifier + data.skills.${skill}.proficiency * data.proficiency_bonus`,
+          character
+        ),
+      })),
+
+      // initiative effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.initiative := data.abilities.dexterity.modifier`,
+          character
+        ),
+      },
+
+      // passive perception effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `data.passive_perception := 10 + data.skills.perception.modifier`,
+          character
+        ),
+      },
+
+      // spell DC effect
+      {
+        name: 'base',
+        priority: 10,
+        ...computeEffect(
+          `spell_dc := 8 + data.scores[data.spellcasting_ability] + data.proficiency_bonus`,
```

Comment:
Naive question: would writing `spell_dc := 8 + data.spell_attack_bonus` work here? Aka having an effect depend on another?

Reply:
Ah! Yes, it would as long as the dependent is defined after the dependency.